test/e2e/azure_test.go

```diff
@@ -466,6 +506,16 @@ var _ = Describe("Workload cluster creation", func() {
 				Args: []string{"--server-side"},
 			}, result)
 
+			By("Verifying that expected Daemonset Pods are running", func() {
+				EnsureDaemonsets(ctx, func() DaemonsetsSpecInput {
```

Comment:
are there any hooks in the e2e framework flow that can allow us to do it as part of create after the worker machines come up? Just like EnsureControlPlaneInitialized allows us to install the Helm charts after the control plane machines are up